From 4bbf06627af302086ac42d1d09a668e4cf21d984 Mon Sep 17 00:00:00 2001
From: Julien Olivain <ju.o@free.fr>
Date: Tue, 21 Dec 2021 22:38:39 +0100
Subject: [PATCH 1796/1936] src/CMakeLists.txt: add an option to disable
 plugins build

Plugins build need the "lv2-ttl-generator" tool. When cross compiling,
the tool binary execution will fail, stopping the build. This patch
add an option to disable plugins to avoid the issue. Cross compilation
is mostly needed for embedded systems. In this case, only the core
binary without GUI is generally needed, without plugins.

To preserve the original behavior, plugins are enabled by default.

Signed-off-by: Julien Olivain <ju.o@free.fr>
---
 src/CMakeLists.txt | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 33abdafb..2900f4b7 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -95,6 +95,7 @@ SET (NoNeonPlease False CACHE BOOL
 SET (PluginLibDir "lib" CACHE STRING
     "Install directory for plugin libraries PREFIX/PLUGIN_LIB_DIR/{lv2,vst}")
 SET (DemoMode FALSE CACHE BOOL "Enable 10 minute silence")
+SET (PluginEnable TRUE CACHE BOOL "Enable Plugins")
 SET (ZynFusionDir "" CACHE STRING "Developers only: zest binary's dir; useful if fusion is not system-installed.")
 mark_as_advanced(FORCE ZynFusionDir)
 
@@ -540,7 +541,9 @@ add_subdirectory(Effects)
 add_subdirectory(Params)
 add_subdirectory(DSP)
 add_subdirectory(Nio)
-add_subdirectory(Plugin)
+if (PluginEnable)
+    add_subdirectory(Plugin)
+endif()
 
 add_library(zynaddsubfx_core STATIC
     version.cpp
-- 
2.47.0

